src/containers/CreateNote/CreateNote.js

```diff
@@ -8,44 +8,58 @@ class CreateNote extends React.Component {
     super(props);
     this.state = {
       title: "",
-      noteItems: [
-        { id: uuid(), text: "default" },
-        { id: uuid(), text: "default" }
-      ]
+      noteItems: [],
+      currentFocus: null
     };
   }
+
   handleChangeTitle = event => {
-    this.setState({ title: event.target.value });
+    this.setState({ title: event.target.value, currentFocus: null });
   };
-  handleChange = event => {
+
+  handleChangeNotes = event => {
     const { noteItems } = this.state;
     let tempNoteList = [...noteItems];
-    const { id, text } = event.target;
+    const { id, value } = event.target;
+
     tempNoteList.forEach(note => {
       if (note.id == event.target.id) {
         note.text = event.target.value;
       }
     });
+
     if (!tempNoteList.find(note => note.id == event.target.id)) {
-      tempNoteList.push({ id, text });
+      tempNoteList.push({ id, text: value });
     }
+
     this.setState({
-      noteItems: tempNoteList
+      noteItems: tempNoteList,
+      currentFocus: id
     });
   };
 
   getListItems() {
     const { noteItems } = this.state;
 
-    let currentList = noteItems.map(item => (
-      <li>
-        <input id={item.id} onChange={this.handleChange} value={item.text} />
-      </li>
-    ));
+    let currentList = noteItems.map(item => {
+      let newNode = (
+        <li key={uuid()}>
+          <input
+            key={uuid()}
+            autoFocus={item.id === this.state.currentFocus}
```

Comment:
`autoFocus` allows an element to be focused after mounting. There was an option to use refs, but I hope this fits our needs without added complexity.

Reply:
Thanks for explaining this to me. autoFocus is built-in. 👍